### PR TITLE
Binary search algorithm 

### PR DESCRIPTION
**Pull request by Udbhav-regadamilli**
Hi @Tezze2001 ,
I have added binary search algorithm to the code base and I want add other different algorithms and data structures too in various programming languages.

I hope this PR is merged

### CODE REVIEW

**Comment by TommasoFerrario18** [comment]
Hi @Udbhav-regadamilli, we are happy to receive your pull request and we are happy to combine your work with ours.  We will accept your pull request as soon as we have set up the entire development environment. Also we are still define how to organize the code.

Our idea is to separate the implementation from the signature of the methods, so that we can create a common interface for each operation, for example: create a list of "abstract" classes and the various implementations in other classes.

Also we are intrested in which programming languages you want to implement other things.

Now we are working on a GitHub action that compiles, executes tests, creates a package and the end update the doc.

File: src/Algorithms/binary_search.cpp [hunk 1]
```diff
@@ -0,0 +1,40 @@
+#include <iostream>
+#include <vector>
+
+using namespace std;
```
Comment (Tezze2001) [inline src/Algorithms/binary_search.cpp]: remove using directive and adjust the source code to make compilable

File: src/Algorithms/binary_search.cpp [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+
+// Function to perform binary search on a sorted vector
+int binarySearch(const vector<int>& arr, int target) {
```
Comment (Tezze2001) [inline src/Algorithms/binary_search.cpp]: use templates in place of int type

File: src/Algorithms/binary_search.cpp [hunk 3]
```diff
@@ -0,0 +1,40 @@
+#include <iostream>
+#include <vector>
+
+using namespace std;
+
```
Comment (Tezze2001) [inline src/Algorithms/binary_search.cpp]: create a new namespace called sorting and put binarySearch function inside

File: src/Algorithms/binary_search.cpp
Comment (Tezze2001) [inline src/Algorithms/binary_search.cpp:1]: - [ ] rename Algorithms folder in sorting
- [ ] rename binary_search.cpp in sorting.cpp, at the end you should have " ./src/sorting/sorting.cpp "
- [ ] create a file header called ./src/sorting/sorting.h and put inside the sing of binarySearch function (Attection: when you create header file you should managed double referencing using #ifndef macro)
- [ ] define a new namespace inside  ./src/sorting/sorting.h  and  ./src/sorting/sorting.cpp called sorting (name of the folder) and put the code inside

File: src/Algorithms/binary_search.cpp [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+        // If target is smaller, ignore right half
+        else
+            right = mid - 1;
+    }
+
+    // If target is not found in the array
+    return -1;
+}
+
+int main() {
```
Comment (Tezze2001) [inline src/Algorithms/binary_search.cpp]: move the main function inside ./tests folder

**Review by Tezze2001 — CHANGES_REQUESTED** [review]

**Comment by Tezze2001** [comment]
If you want continue to contribute, we can talk together on Discord, in this way we can easily explain our goals!

Thank you for your work 🥇 !

**Comment by Udbhav-regadamilli** [comment]
Hi @Tezze2001 ,

I think I have made all the specified changes with the code base.

- [x]  rename Algorithms folder in sorting
- [x]  rename binary_search.cpp in sorting.cpp, at the end you should have " ./src/sorting/sorting.cpp "
- [x]  create a file header called ./src/sorting/sorting.h and put inside the sing of binarySearch function (Attection: when you create header file you should managed double referencing using #ifndef macro)
- [x] define a new namespace inside ./src/sorting/sorting.h and ./src/sorting/sorting.cpp called sorting (name of the folder) and put the code inside

And yes I want to continue contributing.

**Comment by Tezze2001** [comment]
I can see only test file but not the other changes

**Comment by Udbhav-regadamilli** [comment]
Now I have committed the changes, please verify them.

**Review by Tezze2001 — APPROVED** [review]
Perfect, thank you!

You can implement everything written in a issue, note that @TommasoFerrario18 is implementing linear data structures.

**Pull request merged**